plugins/Referrers/API.php

```diff
@@ -365,6 +365,35 @@ public function getSocials($idSite, $period, $date, $segment = false, $expanded
 
         $dataTable = Archive::createDataTableFromArchive(Archiver::SOCIAL_NETWORKS_RECORD_NAME, $idSite, $period, $date, $segment, $expanded, $flat);
 
+        $groupIntagramLabel = function($dataTable) use (&$groupIntagramLabel) {
```

Comment:
@sgiehl might be better to create a filter for this? would remove some complexity and be better to read etc. like `$dataTable->filter('Piwik\Plugins\Referrers\DataTable\Filter\Group')` then you don't need to check if it's a map etc and in theory it would be even possible to write a unit test for the group filter but wouldn't really be needed.